src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -345,6 +345,13 @@ protected string ParseServiceName(ConfigurationKeyValue kv)
 			return ConfigConsts.DefaultValues.UnknownServiceName;
 		}
 
+		protected string ParseEnvironment(ConfigurationKeyValue kv)
+		{
+			if (kv == null || string.IsNullOrEmpty(kv.Value)) return null;
```

Comment:
This means that it will impossible to configure empty string as environment name - is this indeed what we want to do?

Reply:
All other values check to null or empty. This is the reason why I also use such validation. Also, I don't see the point to use empty string as environment.